hail_search/hail_search_query.py

```diff
@@ -139,16 +146,19 @@ def _enum_field(value, enum, ht_globals=None, annotate_value=None, format_value=
 
         return value
 
-    def __init__(self, data_type, sample_data, genome_version, sort=XPOS, num_results=100, **kwargs):
+    def __init__(self, data_type, sample_data, genome_version, sort=XPOS, num_results=100, inheritance_mode=None, **kwargs):
         self._genome_version = genome_version
         self._sort = sort
         self._num_results = num_results
         self._data_type = data_type
         self._ht = None
+        self._comp_het_ht = None
         self._enums = None
         self._globals = None
+        self._is_recessive_search = inheritance_mode == RECESSIVE
```

Comment:
Would it be better to instead make `inheritance_mode` an attribute and have `_is_recessive_search` be `_has_comp_het_search` `@property`s?